openverse_catalog/dags/providers/provider_workflow_types.py

```diff
@@ -0,0 +1,167 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Dict, Optional, Sequence
+
+
+@dataclass
+class ProviderWorkflow:
+    """
+    Required Arguments:
+
+    dag_id:           string giving a unique id of the DAG to be created.
+    provider_script:  string path for the provider_script file whose main
+                      function is to be run. If the optional argument
+                      `dated` is True, then the function must take a
+                      single parameter (date) which will be a string of
+                      the form 'YYYY-MM-DD'. Otherwise, the function
+                      should take no arguments.
+
+    Optional Arguments:
+
+    default_args:      dictionary which is passed to the airflow.dag.DAG
+                       __init__ method.
+    start_date:        datetime.datetime giving the first valid execution
+                       date of the DAG.
+    max_active_tasks:  integer that sets the number of tasks which can
```

Comment:
This is carried over but just a note for the future - we should probably separate active tasks & active DAG runs into two separate config options. We may have cases where we want to run more than 1 task concurrently within a DAG, but still only have 1 active DAG at a given time.

Reply:
That's a good point! I went ahead and separated them out while I'm here 😄